torchaudio/functional.py

```diff
@@ -419,21 +419,90 @@ def DB_to_amplitude(
     return ref * torch.pow(torch.pow(10.0, 0.1 * x), power)
 
 
+def _hz_to_mel(freq: Tensor, mel_scale: str = "htk") -> Tensor:
+    r"""Convert Hz to Mels
+
+    Args:
+        freqs (Tensor): Frequencies in Hz
+        mel_scale (str, optional): Scale to use: ``htk`` or ``slaney``. (Default: ``htk``)
+
+    Returns:
+        mels (Tensor): Input frequencies in Mels
+    """
+
+    if mel_scale not in ['slaney', 'htk']:
+        raise ValueError('mel_scale should be one of "htk" or "slaney".')
+
+    if mel_scale == "htk":
+        return 2595.0 * torch.log10(1.0 + (freq / 700.0))
```

Comment:
I wonder if there's a way of rearranging the calculation to improve numerical stability. We know that all values will be greater than 1, if that helps.

Reply:
Trying `torch.log10((700. + freq) / 700.)`.

Reply:
What are typical ranges on values within freq and do our tests mimic those adequately?

Reply:
Some hints [here](https://github.com/librosa/librosa/blob/master/tests/test_filters.py#L76) and [here](https://librosa.github.io/librosa/generated/librosa.filters.mel.html). The former aligns with the tests we already have. Others to suggest?

Reply:
`torch.log10((700. + freq) / 700.)` behaves better :)

Reply:
I wouldn't necessarily just rely on the librosa tests. We should understand how this is used for our own sake and to have something to point to when it fails outside of those domains.